src/purview.ts

```diff
@@ -1189,57 +914,332 @@ const globalStateTrees: Record<string, StateTree | undefined> = {}
+        }
+        if (req.purviewCSSRendered) {
+          throw new Error(RENDER_CSS_ORDERING_ERROR)
+        }
+        root = {
+          connected: false,
+          cssState: req.purviewCSSState,
+          onError: onUnseenError,
+        }
+      }
+
+      const pNode = await renderComponent(component, component._id, root)
+      if (!pNode) {
+        throw new Error("Expected non-null node")
+      }
 
-  async deleteStateTree(id: string): Promise<void> {
-    delete globalStateTrees[id]
-  },
+      if (purviewState) {
+        return ""
+      } else {
+        await reloadOptions.saveStateTree(
+          component._id,
+          makeStateTree(component, false),
+        )
+        return toHTML(pNode)
+      }
+    })
+  }
 
-  async saveCSSState(id: string, cssState: CSSState): Promise<void> {
-    globalCSSState[id] = cssState
-    if (process.env.NODE_ENV !== "test") {
-      setTimeout(() => this.deleteCSSState(id), DELETE_INTERVAL)
+  export async function renderCSS(req: http.IncomingMessage): Promise<string> {
+    const cssState = req.purviewCSSState
+    if (!cssState) {
+      req.purviewCSSRendered = true
+      return ""
     }
-  },
 
-  async getCSSState(id: string): Promise<CSSState | null> {
-    return globalCSSState[id] ?? null
-  },
+    const { id, cssRules } = cssState
+    const textPNode = createTextPNode(cssRules.join("\n"))
+    const pNode = createPNode(
+      "style",
+      { id: STYLE_TAG_ID, "data-css-state-id": id },
+      [textPNode],
+    )
+
+    cssState.nextRuleIndex = cssRules.length
+    await reloadOptions.saveCSSState(id, cssState)
+    req.purviewCSSRendered = true
+    return toHTML(pNode)
+  }
+  export import JSX = JSXInternal
+
+  export const reloadOptions = {
+    async saveStateTree(id: string, tree: StateTree): Promise<void> {
+      globalStateTrees[id] = tree
+      if (process.env.NODE_ENV !== "test") {
+        setTimeout(() => this.deleteStateTree(id), DELETE_INTERVAL)
+      }
+    },
+
+    async getStateTree(id: string): Promise<StateTree | null> {
+      return globalStateTrees[id] ?? null
+    },
+
+    async deleteStateTree(id: string): Promise<void> {
+      delete globalStateTrees[id]
+    },
+
+    async saveCSSState(id: string, cssState: CSSState): Promise<void> {
+      globalCSSState[id] = cssState
+      if (process.env.NODE_ENV !== "test") {
+        setTimeout(() => this.deleteCSSState(id), DELETE_INTERVAL)
+      }
+    },
+
+    async getCSSState(id: string): Promise<CSSState | null> {
+      return globalCSSState[id] ?? null
+    },
 
-  async deleteCSSState(id: string): Promise<void> {
-    delete globalCSSState[id]
-  },
+    async deleteCSSState(id: string): Promise<void> {
+      delete globalCSSState[id]
+    },
+  }
+  export const scriptPath = pathLib.resolve(
+    __dirname,
+    "..",
+    "dist",
+    "bundle",
+    "browser.js",
+  )
+  // tslint:disable-next-line
+  export abstract class Component<P, S> extends InternalComponent<P, S> {}
```

Comment:
This bit of inelegance and the change on line 1238 below are necessary to specify the correct type for `Component` within the `Purview` namespace AND populate it with the real value of `Component` exported by `component.ts`

It dawns on me that it is possible line 1238 could be removed, since the `extends` statement may be sufficient.